modin/pandas/indexing.py

```diff
@@ -528,6 +528,8 @@ def __getitem__(self, key):
             else:
                 result_slice = self.df.columns.slice_locs(col_loc.start, col_loc.stop)
                 return self.df.iloc[:, slice(*result_slice)]
+        if self.df.empty:
+            return self.df._default_to_pandas(lambda df: df.loc[key])
```

Comment:
maybe we should put this at the top of the function, so we make sure that the execution won't go on that probably-failing fast path above?

Reply:
Yes, good idea.